### PR TITLE
revise connection pool cleanup logic by reducing unncessary loop,

### PR DESCRIPTION
**Pull request by yaojuncn**
some details from
https://github.com/brettwooldridge/HikariCP/issues/379
based on original pull request
https://github.com/brettwooldridge/HikariCP/pull/383

### CODE REVIEW

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java [hunk 1]
```diff
@@ -631,17 +631,20 @@ public void run()
...
-         for (PoolBagEntry bagEntry : bag) {
-            if (removable > 0 && idleTimeout > 0L && clockSource.elapsedMillis(bagEntry.lastAccess, now) > idleTimeout) {
-               if (connectionBag.reserve(bagEntry)) {
-                  closeConnection(bagEntry, "(connection passed idleTimeout)");
-                  removable--;
-               }
-            }
-         }
-         
+          if(idleTimeout > 0L) {
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: Space between 'if' and parenthesis.  Indentation is 3 spaces, not 4.

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java [hunk 2]
```diff
@@ -631,17 +631,20 @@ public void run()
...
+              final List<PoolBagEntry> bag = connectionBag.values(STATE_NOT_IN_USE);
+              int removable = bag.size() - config.getMinimumIdle();
+              if (removable > 0) {
```
Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: if (removable > 0) check needs to be inside for loop too

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java [hunk 3]
```diff
@@ -631,17 +631,20 @@ public void run()
...
+                  for (PoolBagEntry bagEntry : bag) {
+                      if (clockSource.elapsedMillis(bagEntry.lastAccess, now) > idleTimeout) {
```
Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: please remove outer if (removable > 0) and add it here:
if (removable > 0 && clockSource.elapsedMillis(bagEntry.lastAccess, now) > idleTimeout)

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java [hunk 4]
```diff
@@ -631,17 +631,21 @@ public void run()
          previous = now;
 
          logPoolState("Before cleanup\t");
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: Go ahead and move the before cleanup and after cleanup log inside of the ``if (idleTimeout < 0)`` conditional.

**Comment by yaojuncn** [comment]
Btw, although it looks to me the last line of fillPool() can also be put into the if (idleTimeout < 0) part because ideally that would not be needed to fillPool() from the housekeeper any more; 
personally I'd prefer to keep it outside just in case; and there's actually no harm to be there;

**Comment by brettwooldridge** [comment]
The ``fillPool()`` call still needs to be there.  Connections can be retired in a number of ways and need to be replaced.

For example, a separate task in the scheduler executes when a connection reaches its maxLifetime.  User's can also explicitly evict connections.  And finally, certain SQLExceptions from the database can result in a connection being evicted from the pool.

**Comment by yaojuncn** [comment]
I see; Will read your code more for internals:)

**Comment by brettwooldridge** [comment]
For quick reference, the maxLifetime retirement occurs [here](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/pool/PoolBagEntry.java#L87), user evictions occur [here](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/pool/PoolBagEntry.java#L87) and [here](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/pool/HikariPool.java#L413), and SQLException related evictions occur [here](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/proxy/ConnectionProxy.java#L105).

**Comment by yaojuncn** [comment]
Thx

**Pull request merged**